test/src/helper/mock/tendermintMessage.ts

```diff
@@ -0,0 +1,257 @@
+                    type: "consensusmessage",
+                    messages: decoded[1].map((d: any) => {
+                        const inner = RLP.decode(d);
+                        return ({
+                            on: {
+                                step: {
+                                    height: readUIntRLP(inner[0][0][0]),
+                                    view: readUIntRLP(inner[0][0][1]),
+                                    step: readUIntRLP(inner[0][0][2]) as Step,
+                                },
+                                blockHash: readOptionalRlp(
+                                    inner[0][1],
+                                    (buffer) => new H256(buffer.toString("hex"))
+                                ),
+                            },
+                            signature: inner[1].toString("hex"),
+                            signerIndex: readUIntRLP(inner[2]),
+                        })
+                    }),
+                };
+                break;
+            }
+            case MessageType.MESSAGE_ID_PROPOSAL_BLOCK: {
+                message = {
+                    type: "proposalblock",
+                    signature: decoded[1].toString("hex"),
+                    view: readUIntRLP(decoded[2]),
+                    message: uncompressSync(decoded[3]),
+                };
+                break;
+            }
+            case MessageType.MESSAGE_ID_STEP_STATE: {
+                message = {
+                    type: "stepstate",
+                    voteStep: {
+                        height: readUIntRLP(decoded[1][0]),
+                        view: readUIntRLP(decoded[1][1]),
+                        step: readUIntRLP(decoded[1][2]) as Step,
+                    },
+                    proposal: readOptionalRlp(decoded[2], (buffer) =>  new H256(buffer.toString("hex"))),
+                    lockView: readOptionalRlp(decoded[3], readUIntRLP),
+                    knownVotes: decoded[4],
+                };
+                break;
+            }
+            case MessageType.MESSAGE_ID_REQUEST_MESSAGE: {
+                message = {
+                    type: "requestmessage",
+                    voteStep: {
+                        height: readUIntRLP(decoded[1][0]),
+                        view: readUIntRLP(decoded[1][1]),
+                        step: readUIntRLP(decoded[1][2]) as Step,
+                    },
+                    requestedVotes: decoded[2],
+                };
+                break;
+            }
+            case MessageType.MESSAGE_ID_REQUEST_PROPOSAL: {
+                message = {
+                    type: "requestproposal",
+                    height: readUIntRLP(decoded[1]),
+                    view: readUIntRLP(decoded[2]),
+                };
+                break;
+            }
+            default: {
+                throw new Error(`Unexpected message id ${id}`);
+            }
+        }
+        Emitter.emit(message.type, message);
```

Comment:
`emit` in `fromBytes` is awkward.

Reply:
I agree, but the other part of `mock` is implemented like this.
I think refactoring this should be in separated PR since it's too large to fit in this one.